internal/services/authorization/pim_eligible_role_assignment_resource.go

```diff
@@ -653,3 +653,10 @@ func deleteEligibilityRoleAssignmentSchedule(ctx context.Context, client *roleel
 		return result, "Deleted", nil
 	}
 }
+
+func waitTimeoutFromCtx(ctx context.Context, defaultValue time.Duration) time.Duration {
```

Comment:
this default value will never get used - `ctx`'s should _always_ be wrapped by a timeout, so can we remove this method and instead use:

```go
deadline, ok := ctx.Deadline()
if !ok {
  return fmt.Errorf("internal-error: context had no deadline")
}

// ...
Timeout: time.Until(deadline),
```

Reply:
that makes seanse. updated in this way.